File: library/src/main/java/cz/developer/library/ui/switchs/DebugSwitchFragment.java
package cz.developer.library.ui.switchs;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.Switch;

import com.quant.titlebar.TitleBarFragment;

import java.util.Map;

import cz.developer.library.DeveloperManager;
import cz.developer.library.R;


/**
 * Created by cz on 15/12/1.
 * debug信息
 */
public class DebugSwitchFragment extends TitleBarFragment {
    private LinearLayout container;
    private String title;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Bundle arguments = getArguments();
        if (null != arguments) {
            title = arguments.getString("title");
        }
    }

    @Nullable
    @Override
    public View onCreateView(Context context, LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_debug_switch, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setTitleText(title);
        setOnBackClickListener(v->getFragmentManager().popBackStack());

        container= (LinearLayout) view.findViewById(R.id.ll_container);
        //添加附加选项
        ISwitchInterface switchInterface = DeveloperManager.getInstances().getSwitchInterface();
        if(null!=switchInterface){
            Map<String, String> extrasItems = switchInterface.getExtrasItems();
            if(null!=extrasItems){
                int padding = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP,12,getResources().getDisplayMetrics());
                for(Map.Entry<String,String> entry:extrasItems.entrySet()){
                    Switch switchView=new Switch(getContext());
                    switchView.setText(entry.getKey());
                    switchView.setPadding(padding,padding,padding,padding);
                    switchView.setChecked(switchInterface.itemIsChecked(entry.getValue()));
                    switchView.setBackgroundResource(R.drawable.white_item_selector);
                    switchView.setOnCheckedChangeListener((buttonView, isChecked) ->switchInterface.onSwitchItemCheckedChanged(entry.getValue(),isChecked));
                    container.addView(switchView, LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.WRAP_CONTENT);
                }
            }
        }
    }
}

File: library/src/main/java/cz/developer/library/switchs/DebugSwitchFragment.kt
package cz.developer.library.ui.switchs

import android.graphics.Color
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutCompat
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Switch
import cz.developer.library.*
import cz.developer.library.prefs.DeveloperPrefs

import kotlinx.android.synthetic.main.fragment_debug_switch.*


/**
 * Created by cz on 15/12/1.
 * debug信息
 */
internal class DebugSwitchFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_debug_switch, container, false)
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val activity=activity
        if(activity is AppCompatActivity){
            toolBar.title = arguments?.getString("title")
            toolBar.subtitle=arguments?.getString("desc")
            activity.setSupportActionBar(toolBar)
            activity.supportActionBar?.setDisplayHomeAsUpEnabled(true)
            toolBar.setNavigationOnClickListener{ fragmentManager.popBackStack() }
        }
        //添加附加选项
        val switchInterface = DeveloperManager.developerConfig.switchItem
        val items= mutableListOf<DeveloperConfig.SwitchItem.Item>()
        items.add(DeveloperConfig.SwitchItem.Item(DeveloperPrefs.HIERARCHY_KEY,"视图附加功能"))
        if(null!=switchInterface?.items){
            items.addAll(switchInterface?.items)
        }
        val padding = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 16f, resources.displayMetrics).toInt()
        items.forEachIndexed { index,item->
            val key=item.key?:return
            //是否选中
            val switchView = Switch(context)
            switchView.text = item.desc
            switchView.isChecked=DeveloperPrefs.getBoolean(key)
            switchView.setTextColor(Color.BLACK)
            switchView.setTextSize(TypedValue.COMPLEX_UNIT_SP,16f)
            switchView.setPadding(padding, padding, padding, padding)
            switchView.setBackgroundResource(R.drawable.developer_white_item_selector)
            switchView.setOnCheckedChangeListener { _, isChecked ->
                if(0<index){
                    switchInterface?.itemChecked?.invoke(key,isChecked)
                } else {
                    //开启/关闭视图调试
                    DeveloperPrefs.setBoolean(key,isChecked)
                    DeveloperActivityManager.forEach {
                        if(isChecked){
                            it.openDeveloperLayout()
                        } else {
                            it.closeDeveloperLayout()
                        }
                    }
                }
            }
            menuContainer.addView(switchView, LinearLayoutCompat.LayoutParams.MATCH_PARENT, LinearLayoutCompat.LayoutParams.WRAP_CONTENT)
        }
    }
}
